diff --git a/src/tests/runtest b/src/tests/runtest
index 036dc771..e506251e 100755
--- a/src/tests/runtest
+++ b/src/tests/runtest
@@ -24,6 +24,7 @@
 : ${DISABLE_GUI_TESTS:=''}
 : ${DISABLE_DAEMONIZE_IN_TESTS:=''}
 
+pwd
 BUS_REQUIRED_TESTS="
 ibus-bus
 ibus-config
@@ -135,6 +136,13 @@ run_test_case()
         func_copy_component "../$top_srcdir/engine/simple.xml"
         func_copy_component "../$top_srcdir/conf/memconf/memconf.xml"
 
+        cat components/memconf.xml
+        (cd ../../../; pwd)
+        ls ../../../conf/memconf
+        file ../../../conf/memconf/ibus-memconf
+        bash -x ../../../conf/memconf/ibus-memconf
+        strace ../../../conf/memconf/.libs/ibus-memconf
+
         IBUS_COMPONENT_PATH=$PWD/components
         export IBUS_COMPONENT_PATH
 
